### ISSUE TITLE
Drawing a spritebatch causes the next drawn object to blink

### ISSUE THREAD

**Issue by miguelggcc** [ggez/ggez#1025]
When drawing something after a spritebatch, the object seems to blink or rapidly move out of position. This only seems to happen if I draw the spritebatch with  ``dest()`` other than [0.0, 0.0].

Here's a a repro:

```
use ggez::event;
use ggez::graphics;
use ggez::graphics::DrawParam;
use ggez::{Context, GameResult};

struct MainState {
    sb: graphics::spritebatch::SpriteBatch,
    idx: graphics::spritebatch::SpriteIdx,
    mesh: graphics::Mesh,
}

impl MainState {
    fn new(ctx: &mut Context) -> GameResult<MainState> {
        let img = graphics::Image::new(ctx, "/rainbow.png")?;
        let mut sb = graphics::spritebatch::SpriteBatch::new(img);
        let idx = sb.add(graphics::DrawParam::default());
        let mesh = graphics::MeshBuilder::new()
            .rectangle(
                graphics::DrawMode::fill(),
                graphics::Rect::new(100.0, 100.0, 100.0, 100.0),
                graphics::Color::WHITE,
            )?
            .build(ctx)?;
        let s = MainState { sb, idx, mesh };
        Ok(s)
    }
}

impl event::EventHandler for MainState {
    fn update(&mut self, _ctx: &mut Context) -> GameResult {
        Ok(())
    }

    fn draw(&mut self, ctx: &mut Context) -> GameResult {
        graphics::clear(ctx, graphics::Color::BLACK);
        let draw_param = graphics::DrawParam::new();
        self.sb.set(self.idx, draw_param)?;
        graphics::draw(ctx, &self.sb, draw_param.dest([200.0, 200.0]))?;
        graphics::draw(ctx, &self.mesh, DrawParam::new().offset([0.0,0.0]))?;
        graphics::present(ctx)?;
        Ok(())
    }
}
pub fn main() -> GameResult {
    let cb = ggez::ContextBuilder::new("example", "ggez").resources_dir_name("resources");
    let (mut ctx, event_loop) = cb.build()?;
    let state = MainState::new(&mut ctx)?;
    event::run(ctx, event_loop, state)
}

```
And here's what happens:

https://user-images.githubusercontent.com/100235899/161107568-03bac454-c29f-4b78-a5a0-80912a337125.mp4

I'm using:
 - ggez version:  release 0.7.0,
 - OS: Windows 10,

**Comment by PSteinhaus** [comment]
This is really weird, thanks for reporting.
Sadly I can't reproduce this behavior. For me the rectangle is always at its proper position on the top left of the sprite... changing the `DrawParam` of the sprite or the spritebatch as a whole doesn't seem to change anything.

Does commenting out
```rust
self.sb.set(self.idx, draw_param)?;
```
change anything for you? Probably not, but I'd like to know anyway.

If it's ok to you, could you try to adapt your repro to the [wgpu PR that is currently open](https://github.com/ggez/ggez/pull/1020) and see if the issue persists there as well? My assumption would be that that shouldn't be the case, as the whole graphics stuff is rewritten completely there, but the whole API changes quite a bit with it as well, so some changes would be necessary.
I'll try to edit my comment and add an adapted version later, but it shouldn't be that much. `SpriteBatch` has been replaced by `InstanceArray` and you now need to always draw on a `Canvas`, usually created from the frame image, but that should be the greatest changes.

**Comment by miguelggcc** [comment]
Weird, I forced the game to use the dedicated GPU because for some reason only the integrated Intel GPU was being used and now the problem is gone

**Comment by nobbele** [comment]
Intel graphics are frequently buggy

**Comment by PSteinhaus** [comment]
Hmm, interesting.
Did you try the wgpu fork too? I'd be curious whether the issue persists there. The code I've given in my previous comment needs to be slightly adapted now, as `draw_instances` and `draw_mesh` have been replaced by just `draw`.

**Comment by nobbele** [comment]
Stale issue probably fixed in 0.8.0, please re-open if it persists
